arangod/V8Server/V8DealerFeature.cpp

```diff
@@ -361,6 +352,61 @@ void V8DealerFeature::start() {
   startGarbageCollection();
 }
 
+void V8DealerFeature::copyInstallationFiles() {
+  // get base path from DatabasePathFeature
+  auto dbPathFeature = application_features::ApplicationServer::getFeature<DatabasePathFeature>();
+  const std::string copyJSPath = FileUtils::buildFilename(dbPathFeature->directory(), "js");
+  if (copyJSPath == _startupDirectory) {
+    LOG_TOPIC(FATAL, arangodb::Logger::V8)
+    << "'javascript.startup-directory' cannot be inside 'database.directory'";
+    FATAL_ERROR_EXIT();
+  }
+  TRI_ASSERT(!copyJSPath.empty());
+  
+  const std::string checksumFile = FileUtils::buildFilename(_startupDirectory, StaticStrings::checksumFileJs);
+  const std::string copyChecksumFile = FileUtils::buildFilename(copyJSPath, StaticStrings::checksumFileJs);
+  
+  bool overwriteCopy = false;
+  if (!FileUtils::exists(copyJSPath) ||
+      !FileUtils::exists(checksumFile) ||
+      !FileUtils::exists(copyChecksumFile)) {
+    overwriteCopy = true;
+  } else {
+    try {
+      overwriteCopy = (FileUtils::slurp(copyChecksumFile) != FileUtils::slurp(checksumFile));
+    } catch(basics::Exception const& e) {
+      LOG_TOPIC(ERR, Logger::V8) << "Error reading '" << StaticStrings::checksumFileJs <<
+      "' from disk: " << e.what();
+      overwriteCopy = true;
+    }
+  }
+  
+  if (overwriteCopy) {
+    LOG_TOPIC(DEBUG, Logger::V8) << "Copying JS installation files to '" << copyJSPath << "'";
+    int res = TRI_ERROR_NO_ERROR;
+    if (FileUtils::exists(copyJSPath)) {
+      res = TRI_RemoveDirectory(copyJSPath.c_str());
+      if (res != TRI_ERROR_NO_ERROR) {
+        LOG_TOPIC(FATAL, Logger::V8) << "Error cleaning JS installation path '"
+          << TRI_errno_string(res) << "'";
+        FATAL_ERROR_EXIT();
```

Comment:
Please change to something that prints both the path and the error message. That will make it easier to debug later:
```
    LOG_TOPIC(FATAL, Logger::V8) << "Error cleaning JS installation path '" << copyJSPath << "': " << TRI_errno_string(res);
```